### PR TITLE
Testing improvements

### PR DESCRIPTION
**Pull request by LouieK22**
- Added new tests for empty returns & __tostring inheritance
- Error testing now uses a matrix

### CODE REVIEW

File: src/test.ts [hunk 1]
```diff
@@ -6,6 +6,14 @@ import { TranspilerError, TranspilerErrorType } from "./class/errors/TranspilerE
 
 require("mocha");
 
+interface ErrorMatix {
```
Comment (osyrisrblx) [inline src/test.ts]: typo, should be `ErrorMatrix`

File: tests/src/function.spec.ts [hunk 2]
```diff
@@ -13,6 +13,34 @@ export = () => {
...
+			} else {
+				return;
+			}
+		}
+
+		expect(doStuff(true)).to.equal("cleans room");
+		expect(doStuff(false)).never.to.be.ok();
+	});
+
+	/* https://github.com/roblox-ts/roblox-ts/issues/114
```
Comment (osyrisrblx) [inline tests/src/function.spec.ts]: This test is actually okay. The broken syntax was having the variables created before destructuring assignment.

i.e.
```TS
let a, b;
[a, b] = foo();
```

File: tslint.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
 		"no-unnecessary-qualifier": true,
 		"no-unused-variable": true,
 		"no-var-keyword": true,
+		"object-literal-sort-keys": false,
```
Comment (osyrisrblx) [inline tslint.json]: why was this added?

Comment (LouieK22) [reply tslint.json]: tslint complains about objects not being alphabetically sorted otherwise, which makes the matrices for error testing a bit more unclear.
